research/blueprint_algo/blueprint_short_deck_poker.py

```diff
@@ -79,33 +58,46 @@
 utils.random.seed(42)
 
 
+# TODO: In general, wondering how important this function is if we are to use the blueprint algo for more than the
+#  preflop round? Would using just sigma allow for a more complete rendering of strategies for infosets?
 def update_strategy(state: ShortDeckPokerState, i: int):
     """
 
-    :param rs: realstate, a tuple of two ints, first is card for player one, second player 2
-    :param h: the action sequences without the card information
+    :param state: the game state
     :param i: the player, i = 1 is always first to act and i = 2 is always second to act, but they take turns who
         updates the strategy (only one strategy)
     :return: nothing, updates action count in the strategy of actions chosen according to sigma, this simple choosing of
         actions is what allows the algorithm to build up preference for one action over another in a given spot
     """
-    ph = 2 if len(state.h) == 1 else 1  # this is always the case no matter what i is
+    ph = state.player_i  # this is always the case no matter what i is
 
-    if (
-        state.is_terminal
-    ):  # or if p_i is not in the hand or if betting round is > 0, strategy is only
-        return  # updated in betting round 1 for Pluribus, but I am doing all rounds in this example
-    # elif h is chance_node:  -- we don't care about chance nodes here, but we will for No Limit
+    print(i, ph)
+
+    print(state._table.players[i]._is_active)
+
+    if state.is_terminal or state._table.players[i]._is_active is False:
```

Comment:
This is a little bit ugly, ideally we don't access private state outside of a class, I suggested a new `players` property for the state class, then we can change to this:
```python
if state.is_terminal or not state.players[i].is_active:
```

Reply:
As we should be able to access `is_active`, not the private `_is_active` too!

Reply:
Yes excuse my ad hoc debugging!